src/components/views/rooms/AppsDrawer.js

```diff
@@ -62,13 +63,13 @@ export default class AppsDrawer extends React.Component {
 
     componentDidMount() {
         ScalarMessaging.startListening();
-        WidgetStore.instance.on(this.props.room.roomId, this._updateApps);
+        WidgetLayoutStore.instance.on(WidgetLayoutStore.emissionForRoom(this.props.room), this._updateApps);
```

Comment:
If this component can receive a new room through a props change, then you'll need to handle swapping this listener off the old room and onto the new room.

Reply:
Not sure I understand - this isn't anything different than before?

Reply:
Ah, I didn't really look at the previous line here... I'm saying I would expect `componentDidUpdate` to swap listeners to the new room ID if it's possible it can change. Since you're right it would have been an issue before, I guess it means we don't actually change rooms after the first mount here.